server/aesdsocket.c

```diff
@@ -307,13 +351,15 @@ void* receive_send_data(void* thread_param)
 	// close connection
 	close(thread_func_args->client_fd);
 	//syslog(LOG_DEBUG, "Closed connection from %s", ipstr);
+	close(write_fd);
 	free(thread_func_args->buffer);
 	free(thread_func_args->transmit_buffer);
 	thread_func_args->thread_complete_success = true;
 	return NULL;	
 }
 
 //add timespecs and store in result
+#ifndef USE_AESD_CHAR_DEVICE
```

Comment:
@midhun9 think about how you could have less code in #ifdefs, continue to compile the timestamp code and only #ifdef the code which starts the timer.